lib/galaxy/model/__init__.py

```diff
@@ -2972,6 +2974,43 @@ class GenomeIndexToolData(Base, RepresentById):  # TODO: params arg is lost
     user: Mapped[Optional["User"]] = relationship()
 
 
+class ChatExchange(Base, RepresentById):
+
+    __tablename__ = "chat_exchange"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    user_id: Mapped[int] = mapped_column(ForeignKey("galaxy_user.id"), index=True, nullable=False)
+    job_id: Mapped[Optional[int]] = mapped_column(ForeignKey("job.id"), index=True, nullable=True)
+
+    user: Mapped["User"] = relationship(back_populates="chat_exchanges")
+    messages: Mapped[List["ChatExchangeMessage"]] = relationship(back_populates="chat_exchange", cascade_backrefs=False)
+
+    def __init__(self, user, job_id=None, message=None, **kwargs):
+        self.user = user
+        self.job_id = job_id
+        self.messages = []
+        if message:
+            self.add_message(message)
+
+    def add_message(self, message):
+        self.messages.append(ChatExchangeMessage(message=message))
+
+
+class ChatExchangeMessage(Base, RepresentById):
+    __tablename__ = "chat_exchange_message"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    chat_exchange_id: Mapped[int] = mapped_column(ForeignKey("chat_exchange.id"), index=True)
+    create_time: Mapped[datetime] = mapped_column(default=now, nullable=True)
+    message: Mapped[str] = mapped_column(Text)
+    feedback: Mapped[int] = mapped_column(Integer, nullable=True)
```

Comment:
Can feedback be `None`? If so, we can use `Optional[Mapped[int]]` and drop `nullable=True`.